pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -0,0 +1,144 @@
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(flowcontrol.GroupName, legacyscheme.Scheme, legacyscheme.ParameterCodec, legacyscheme.Codecs)
+
+	if apiResourceConfigSource.VersionEnabled(flowcontrolv1alpha1.SchemeGroupVersion) {
+		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolv1alpha1.SchemeGroupVersion.Version] = p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter)
+	}
+	return apiGroupInfo, true
+}
+
+func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) map[string]rest.Storage {
+	storage := map[string]rest.Storage{}
+
+	// flow-schema
+	flowSchemaStorage, flowSchemaStatusStorage := flowschemastore.NewREST(restOptionsGetter)
+	storage["flowschemas"] = flowSchemaStorage
+	storage["flowschemas/status"] = flowSchemaStatusStorage
+
+	// priority-level-configuration
+	priorityLevelConfigurationStorage, priorityLevelConfigurationStatusStorage := prioritylevelconfigurationstore.NewREST(restOptionsGetter)
+	storage["prioritylevelconfigurations"] = priorityLevelConfigurationStorage
+	storage["prioritylevelconfigurations/status"] = priorityLevelConfigurationStatusStorage
+
+	return storage
+}
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	systemPreset := SystemPresetData{
+		FlowSchemas:                 bootstrap.SystemFlowSchemas(),
+		PriorityLevelConfigurations: bootstrap.SystemPriorityLevelConfigurations(),
+	}
+	// TODO: default flow-schemas and priority levels
+	return PostStartHookName, systemPreset.EnsureSystemPresetConfiguration(), nil
+}
+
+type SystemPresetData struct {
+	FlowSchemas                 []*flowcontrolv1alpha1.FlowSchema
+	PriorityLevelConfigurations []*flowcontrolv1alpha1.PriorityLevelConfiguration
+}
+
+func (d SystemPresetData) EnsureSystemPresetConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		// Adding system priority classes is important. If they fail to add, many critical system
+		// components may fail and cluster may break.
+		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+			flowcontrolClientSet, err := flowcontrolclient.NewForConfig(hookContext.LoopbackClientConfig)
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
+				return false, nil
+			}
+
+			for _, flowSchema := range d.FlowSchemas {
+				_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						_, err := flowcontrolClientSet.FlowSchemas().Create(flowSchema)
+						if err != nil && !apierrors.IsAlreadyExists(err) {
+							return false, err
+						} else {
+							klog.Infof("created system preset FlowSchema %s", flowSchema.Name)
```

Comment:
if `err != nil && apierrors.IsAlreadyExists(err)` then this code did _not_ create the FlowSchema.

Reply:
not sure if i follow.. we can't create f it already exists, can we?

Reply:
My complaint is that if `err != nil && apierrors.IsAlreadyExists(err)` then this code logs a lie --- it claims to have created the object, but actually something else did.

Reply:
> if err != nil && apierrors.IsAlreadyExists(err) then this code did not create the FlowSchema.

but it's returning a non-nil error from post-start-hook, the apiserver will crash if any hooks failed

Reply:
I am not saying "make it crash", I am saying "do not make it log a lie".  This code could log an accurate message without crashing the apiserver.

Reply:
how about we comment that above the code?

> // returning error here so that post-start-hook fails the kube-apiserver from launching?

Reply:
Sorry, I fear I have not been clear.  I suggest changing line 104 from `} else {` to `} else if err == nil {`.

Reply:
updated